### ISSUE TITLE
Callback: Get metric value per batch

### ISSUE THREAD

**Issue by StefReck** [keras-team/keras#13530]
I'm using the `fit_generator` method, together with a callback to save the values of both the loss and the metric(s) for each batch. I do that by reading from the `logs` dictionary of the on_batch_end method of the callback (e.g. `logs['acc']`). 

With the new class-based metrics, however, only the average of the metric over the whole epoch seems to be accessible in `logs`, not the result of the metric over the last batch (i.e. `logs['acc']` is not the accuracy over just the last batch anymore). The loss `logs['loss']` is still given on a per-batch basis, though, which seems unintuitive to me. While one could use `model.reset_metrics()` on the end of every batch to get batchwise metrics, this will mean that the history object returned by `fit_generator` will also not contain the averaged metric values anymore, but just the one from the last batch.

I therefore think it would be a good idea to also store the values of the metrics over just the last batch in 'logs' of on_batch_end, is there any chance this could get added?

- TensorFlow version:  2.0.0
- Keras version:  2.3.1
